src/ipc/handler.c

```diff
@@ -118,6 +118,9 @@ struct sof_ipc_cmd_hdr *mailbox_validate(void)
 {
 	struct sof_ipc_cmd_hdr *hdr = ipc_get()->comp_data;
 
+	/* reset and clear the stale IPC data */
+	bzero(hdr, SOF_IPC_MSG_MAX_SIZE);
+
```

Comment:
Is this a fix ? it sounds like it should be a sperate "fix" PR. What heppens if this is not done. Is this the local copy of the IPC message or the shared SRAM copy ? if it's SRAM we can race with driver if driver needs to R/W this data ?

Reply:
yes, it should be a separated fix, let me do it. and it is FW local memory and used to store the data copied from the mailbox(shared SRAM memory window).

Reply:
ah, is this because the driver PR is always sending the extended data regardless of whether we have a UUID in topology (and needed for NULL UUID check in FW) ? This wont be needed if we only send UUID when it's in topology.

Reply:
> ah, is this because the driver PR is always sending the extended data regardless of whether we have a UUID in topology (and needed for NULL UUID check in FW) ? This wont be needed if we only send UUID when it's in topology.

Exactly, please see the comment in the kernel PR here https://github.com/thesofproject/linux/pull/2215#discussion_r448724102